lib/Model/Importer/Binary/PECOFFImporter.cpp

```diff
@@ -76,19 +77,46 @@ Error PECOFFImporter::parseProgramHeaders() {
     const object::coff_section *CoffRef = *SecOrErr;
 
     MetaAddress Start = ImageBase + u64(CoffRef->VirtualAddress);
-    Segment Segment({Start, u64(CoffRef->VirtualSize)});
+    model::Segment Segment({Start, u64(CoffRef->VirtualSize)});
 
     Segment.StartOffset = CoffRef->PointerToRawData;
 
     // VirtualSize might be larger than SizeOfRawData (extra data at the end of
     // the section) or viceversa (data mapped in memory but not present in
     // memory, e.g., .bss)
-    Segment.FileSize = std::min(CoffRef->VirtualSize, CoffRef->SizeOfRawData);
+    Segment.FileSize = CoffRef->SizeOfRawData;
+
+    // It is possible that the file size if greather than VirtualSize
+    // because SizeOfRawData is rounded, but VirtSize is not, than
+    // we work it around here.
+    // FIXME: Revisit this.
```

Comment:
How does the rounding works? Can you paste/link docs?

Reply:
VirtualSize represents how many bytes in memory is reserved for the section. SizeOfRawData represents how many many bytes the section has on disk.

VirtualSize can be larger than SizeOfRawData when for example the section has data that we don't store on disk (e.g. .bss).
OTOH, it is possible that VirtualSize < SizeOfRawData, since SizeOfRawData is being rounded to the FileAlignment (512 bytes is the default; also present in the file header https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#optional-header-windows-specific-fields-image-only).

Reply:
> VirtualSize represents how many bytes in memory is reserved for the section. SizeOfRawData represents how many many bytes the section has on disk.
>
> VirtualSize can be larger than SizeOfRawData when for example the section has data that we don't store on disk (e.g. .bss).

OK, this is regular stuff.

> OTOH, it is possible that VirtualSize < SizeOfRawData, since SizeOfRawData is being rounded to the FileAlignment (512 bytes is the default; also present in the file header https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#optional-header-windows-specific-fields-image-only).

So what happens if `VirtualSize < SizeOfRawData`? Are the bytes in the discrepancy mapped into memory or not?
If not, we should shrink `SizeOfRawData`. If yes, we should increase `VirtualSize`.

Reply:
Status?